docs/reference/search/search-your-data/semantic-search-inference.asciidoc

```diff
@@ -0,0 +1,270 @@
+[[semantic-search-inference]]
+=== Tutorial: semantic search with the {infer} API
+++++
+<titleabbrev>Semantic search with the {infer} API</titleabbrev>
+++++
+
+The instructions in this tutorial shows you how to use the {infer} API with the 
+Open AI service to perform semantic search on your data. The following example 
+uses OpenAI's `text-embedding-ada-002` second generation embedding model. You 
+can use any OpenAI models, they are all supported by the {infer} API.
+
+
+[discrete]
+[[infer-openai-requirements]]
+==== Requirements
+
+An https://openai.com/[OpenAI account] is required to use the {infer} API with 
+the OpenAI service. 
+
+
+[discrete]
+[[infer-text-embedding-task]]
+==== Create the inference task
+
+Create the {infer} task by using the <<put-inference-api>>:
+
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/openai_embeddings <1>
+{
+    "service": "openai",
+    "service_settings": {
+        "api_key": "<api_key>" <2>
+    },
+    "task_settings": {
+       "model": "text-embedding-ada-002" <3>
+    }
+}
+------------------------------------------------------------
+// TEST[skip:TBD]
+<1> The task type is `text_embedding` in the path.
+<2> The API key of your OpenAI account. You can find your OpenAI API keys in 
+your OpenAI account under the 
+https://platform.openai.com/api-keys[API keys section]. You need to provide 
+your API key only once. The <<get-inference-api>> does not retrieve your API 
+key.
+<3> The name of the embedding model to use. You can find the list of OpenAI 
+embedding models 
+https://platform.openai.com/docs/guides/embeddings/embedding-models[here].
+
+
+[discrete]
+[[infer-openai-mappings]]
+==== Create the index mapping
+
+The mapping of the destination index - the index that contains the embeddings 
+that the model will create based on your input text - must be created. The 
+destination index must have a field with the <<dense-vector, `dense_vector`>> 
+field type to index the output of the OpenAI model.
+
+[source,console]
+--------------------------------------------------
+PUT openai-embeddings
+{
+  "mappings": {
```

Comment:
The dense_vector field has an [element_type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params) option which defaults to float. For these embeddings which are quite large at 1536 elements let's use the `element_type: byte` option as it requires much less memory